compiler/src/dotty/tools/dotc/core/quoted/PickledQuotes.scala

```diff
@@ -111,4 +107,42 @@ object PickledQuotes {
     }
     tree
   }
+
+  private def classTagToTypeTree(ct: ClassTag[_])(implicit ctx: Context): TypeTree = {
+    val tpe = ct match {
+      case ClassTag.Unit => defn.UnitType
+      case ClassTag.Byte => defn.ByteType
+      case ClassTag.Char => defn.CharType
+      case ClassTag.Short => defn.ShortType
+      case ClassTag.Int => defn.IntType
+      case ClassTag.Long => defn.LongType
+      case ClassTag.Float => defn.FloatType
+      case ClassTag.Double => defn.FloatType
+    }
+    TypeTree(tpe)
+  }
+
+  private def functionAppliedTo(f: Tree, x: Tree)(implicit ctx: Context): Tree = {
+    val x1 = SyntheticValDef(NameKinds.UniqueName.fresh("x".toTermName), x)
+    def x1Ref() = ref(x1.symbol)
+    def rec(f: Tree): Tree = f match {
+      case Block((ddef: DefDef) :: Nil, _: Closure) =>
```

Comment:
Use the `closure` extractor instead (in TreeInfo).

Reply:
`closure` only extracts the `Closure` and here we also need its `DefDef`. I will instead factor out a common extractor from `closureBody` and this me method.